mitmproxy/addons/maplocal.py

```diff
@@ -0,0 +1,104 @@
+import mimetypes
+import re
+import typing
+import urllib
+from pathlib import Path
+from werkzeug.security import safe_join
+
+from mitmproxy import ctx, exceptions, http
+from mitmproxy.addons.modifyheaders import parse_modify_spec, ModifySpec
+
+
+def get_mime_type(file_path: str) -> str:
+    mimetype = (
+        mimetypes.guess_type(file_path)[0]
+        or "application/octet-stream"
+    )
+    return mimetype
+
+
+def file_candidates(url: str, base_path: str) -> typing.List[Path]:
```

Comment:
nit: let's keep accepting a Path here and only `str()` before we `safe_join`. We really want to work on Path objects, hopefully safe_join does support them at some point as well.